api/src/test/java/org/openmrs/api/ProgramWorkflowServiceTest.java

```diff
@@ -740,7 +750,263 @@ public void shouldTestPurgeProgramAttributeType() throws Exception {
                 int totalAttributeTypes = pws.getAllProgramAttributeTypes().size();
 		pws.purgeProgramAttributeType(programAttributeType);
 		assertEquals((totalAttributeTypes - 1), pws.getAllProgramAttributeTypes().size());
-}
+	}
+	
+	
+	@Test
+	public void getPrograms_shouldTestGetPrograms() {
+		List<Program> malPrograms = pws.getPrograms("MAL");
+		List<Program> prPrograms = pws.getPrograms("PR");
+		assertEquals(malPrograms.size(), 1);
+		assertEquals(prPrograms.size(), 3);
+	}
+	
+	@Test
+	public void retireProgram_shouldSetRetiredStateToFalseAndSetAReason() {
+		Program programTest = pws.getAllPrograms().get(0);
```

Comment:
This program doesn't seem to have any associated `ProgramWorkflow`s. Would it be possible to find one that does for this test?

Reply:
No, there are only 3 programs and none of them has workflows. What should I do?

Reply:
Assuming you've found the XML files with data, add a new one with a program and give it some associated workflows. Or just also load the file `org/openmrs/api/include/ProgramWorkflowServiceTest-otherProgramWorkflows.xml` which has a single workflow for Program 2.

Reply:
Ok, I will try. Thank you!

Reply:
Please reach out if you run into any difficulties implementing this!

Reply:
Yes, thank you for support!

Reply:
I loaded the file and it worked. I will upload the commit soon